network.go

```diff
@@ -944,7 +945,10 @@ func (n *network) ipamAllocateVersion(ipVer int, ipam ipamapi.Ipam) error {
 		// irrespective of whether ipam driver returned a gateway already.
 		// If none of the above is true, libnetwork will allocate one.
 		if cfg.Gateway != "" || d.Gateway == nil {
-			if d.Gateway, _, err = ipam.RequestAddress(d.PoolID, net.ParseIP(cfg.Gateway), nil); err != nil {
+			var gatewayOpts = map[string]string{
+				netlabel.Gateway: "",
```

Comment:
@mavenugo I take the blame on this. See https://github.com/docker/libnetwork/pull/707#discussion_r42936310

I am fine to change the value back to "true". This is actually how @rmb938 first wrote it.